### PR TITLE
chore(sso): update to latest keycloak docker setup

### PR DESCRIPTION
**Pull request by megglos**
relates to #OPT-4690

This change is based on https://github.com/oauth2-proxy/oauth2-proxy/tree/master/contrib/local-environment which is a recommended approach (use of a generic oauth proxy) after the previous proxy image by keycloak and it's initial successor, keycloak-gatekeeper https://hub.docker.com/r/keycloak/keycloak-gatekeeper/, was deprecated.

The setup now does not require a docker build anymore, making it simpler as well.

Once merged I can care about updating the 3.6.0 tag

### CODE REVIEW

File: optimize-sso-keycloak-example/docker-compose.yml [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:8080/auth/realms/master"]
+      interval: 5s
+      timeout: 5s
+      retries: 3
+      start_period: 30s
 
   optimize:
-    image: registry.camunda.cloud/optimize-ee/optimize:latest
+    image: registry.camunda.cloud/optimize-ee/optimize:${OPTIMIZE_VERSION:?Please provide the Optimize version as OPTIMIZE_VERSION environment variable}
```
Comment (megglos) [inline optimize-sso-keycloak-example/docker-compose.yml:49]: we could of course automate the example repo release job to put the version here as well, I rather delegated setting the version to the user as they should be aware of the version they want to use, also provided and example in the readme.

File: optimize-sso-keycloak-example/README.md [hunk 2]
```diff
@@ -33,13 +33,12 @@ Put your Optimize license into `./optimize-config/OptimizeLicense.txt`
 
 ## Run docker-compose
 
-1. Build all images with `docker-compose build`
-2. Login to private Camunda Docker EE Registry with `docker login registry.camunda.cloud` Use your EE LDAP credentials to log in.
-3. Start all images `docker-compose up -d`
+1. Login to private Camunda Docker EE Registry with `docker login registry.camunda.cloud`, use your EE LDAP credentials to log in.
+3. Start all images along with providing the Optimize version you want to run with e.g. with `OPTIMIZE_VERSION=3.6.0 docker-compose up -d`
```
Comment (RomanJRW) [inline optimize-sso-keycloak-example/README.md]: 1 -> 3 😉

**Review by RomanJRW — APPROVED** [review]
Kind of just trusting your judgement here but from a high level view the changes look fine to me. One tiny comment for the README, other than that all good ✅

File: optimize-sso-keycloak-example/docker-compose.yml (hunk 1 above)
Comment (ingorichtsmeier) [reply optimize-sso-keycloak-example/docker-compose.yml:49]: Hi @megglos, 
you tag all examples with an optimize version. 
You have to build the plugin code with an optimze version. 
Why not bind the optimize here as well? 
Feels more consitent from a user perspective.

Comment (megglos) [reply optimize-sso-keycloak-example/docker-compose.yml:49]: @ingorichtsmeier the version is managed via maven but docker-compose is isolated from that. The only easy to maintain option I see would be running docker via a maven command would be something like `mvn docker-compose-maven-plugin:up` would that be ok from your perspective?

Comment (ingorichtsmeier) [reply optimize-sso-keycloak-example/docker-compose.yml:49]: Hi @megglos,

I wasn't aware of this. 
Keep it as it is. 

Thank you, Ingo

**Pull request merged**